pkg/render/monitor/monitor.go

```diff
@@ -1003,6 +1032,25 @@ func (mc *monitorComponent) operatorRoleBinding() *rbacv1.RoleBinding {
 			},
 		},
 	}
+
+	secretBinding := &rbacv1.RoleBinding{
+		TypeMeta:   metav1.TypeMeta{Kind: "RoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: CalicoPrometheusOperatorSecret, Namespace: common.TigeraPrometheusNamespace},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      CalicoPrometheusOperator,
+				Namespace: common.TigeraPrometheusNamespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     CalicoPrometheusOperatorSecret,
```

Comment:
Using the same SA since the secrets are created/listed by calico-prometheus-operator